[2024-08-24T12:18:58.319+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-08-24T12:18:58.333+0000] {taskinstance.py:2603} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_airflow_dag.spark_submit_task manual__2024-08-24T11:34:17.052163+00:00 [queued]>
[2024-08-24T12:18:58.340+0000] {taskinstance.py:2603} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_airflow_dag.spark_submit_task manual__2024-08-24T11:34:17.052163+00:00 [queued]>
[2024-08-24T12:18:58.340+0000] {taskinstance.py:2856} INFO - Starting attempt 5 of 5
[2024-08-24T12:18:58.353+0000] {taskinstance.py:2879} INFO - Executing <Task(SparkSubmitOperator): spark_submit_task> on 2024-08-24 11:34:17.052163+00:00
[2024-08-24T12:18:58.358+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=2487) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-08-24T12:18:58.360+0000] {standard_task_runner.py:72} INFO - Started process 2489 to run task
[2024-08-24T12:18:58.360+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_***_dag', 'spark_submit_task', 'manual__2024-08-24T11:34:17.052163+00:00', '--job-id', '67', '--raw', '--subdir', 'DAGS_FOLDER/spark_jobs_dag.py', '--cfg-path', '/tmp/tmpuh15acjt']
[2024-08-24T12:18:58.361+0000] {standard_task_runner.py:105} INFO - Job 67: Subtask spark_submit_task
[2024-08-24T12:18:58.372+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:209: DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
  SQL_ALCHEMY_CONN = conf.get("database", "SQL_ALCHEMY_CONN")

[2024-08-24T12:18:58.401+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_airflow_dag.spark_submit_task manual__2024-08-24T11:34:17.052163+00:00 [running]> on host 62929844d775
[2024-08-24T12:18:58.477+0000] {taskinstance.py:3122} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_***_dag' AIRFLOW_CTX_TASK_ID='spark_submit_task' AIRFLOW_CTX_EXECUTION_DATE='2024-08-24T11:34:17.052163+00:00' AIRFLOW_CTX_TRY_NUMBER='5' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-08-24T11:34:17.052163+00:00'
[2024-08-24T12:18:58.478+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-08-24T12:18:58.478+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-08-24T12:18:58.478+0000] {logging_mixin.py:190} INFO - Current task name:spark_submit_task state:running start_date:2024-08-24 12:18:58.333646+00:00
[2024-08-24T12:18:58.479+0000] {logging_mixin.py:190} INFO - Dag name:spark_***_dag and current dag run status:running
[2024-08-24T12:18:58.479+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-08-24T12:18:58.487+0000] {base.py:84} INFO - Retrieving connection 'spark_local'
[2024-08-24T12:18:58.489+0000] {spark_submit.py:403} INFO - Spark-Submit cmd: spark-submit --master spark://543daafc0f5f:7077 --name arrow-spark --queue root.default --deploy-mode client /opt/bitnami/spark/work/spark_etl_script_docker.py
[2024-08-24T12:19:00.425+0000] {spark_submit.py:579} INFO - python3: can't open file '/opt/bitnami/spark/work/spark_etl_script_docker.py': [Errno 2] No such file or directory
[2024-08-24T12:19:00.435+0000] {spark_submit.py:579} INFO - 24/08/24 12:19:00 INFO ShutdownHookManager: Shutdown hook called
[2024-08-24T12:19:00.437+0000] {spark_submit.py:579} INFO - 24/08/24 12:19:00 INFO ShutdownHookManager: Deleting directory /tmp/spark-a329b42f-ad08-4294-80e1-ccee0cf3efd3
[2024-08-24T12:19:00.493+0000] {taskinstance.py:3301} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://543daafc0f5f:7077 --name arrow-spark --queue root.default --deploy-mode client /opt/bitnami/spark/work/spark_etl_script_docker.py. Error code is: 2.
[2024-08-24T12:19:00.498+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-08-24T12:19:00.498+0000] {logging_mixin.py:190} INFO - Task start:2024-08-24 12:18:58.333646+00:00 end:2024-08-24 12:19:00.497552+00:00 duration:2.163906
[2024-08-24T12:19:00.499+0000] {logging_mixin.py:190} INFO - Task:<Task(SparkSubmitOperator): spark_submit_task> dag:<DAG: spark_***_dag> dagrun:<DagRun spark_***_dag @ 2024-08-24 11:34:17.052163+00:00: manual__2024-08-24T11:34:17.052163+00:00, state:running, queued_at: 2024-08-24 12:18:56.078745+00:00. externally triggered: True>
[2024-08-24T12:19:00.499+0000] {logging_mixin.py:190} INFO - Failure caused by Cannot execute: spark-submit --master spark://543daafc0f5f:7077 --name arrow-spark --queue root.default --deploy-mode client /opt/bitnami/spark/work/spark_etl_script_docker.py. Error code is: 2.
[2024-08-24T12:19:00.499+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=spark_***_dag, task_id=spark_submit_task, run_id=manual__2024-08-24T11:34:17.052163+00:00, execution_date=20240824T113417, start_date=20240824T121858, end_date=20240824T121900
[2024-08-24T12:19:00.514+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-08-24T12:19:00.515+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 67 for task spark_submit_task (Cannot execute: spark-submit --master spark://543daafc0f5f:7077 --name arrow-spark --queue root.default --deploy-mode client /opt/bitnami/spark/work/spark_etl_script_docker.py. Error code is: 2.; 2489)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2995, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3149, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3173, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://543daafc0f5f:7077 --name arrow-spark --queue root.default --deploy-mode client /opt/bitnami/spark/work/spark_etl_script_docker.py. Error code is: 2.
[2024-08-24T12:19:00.542+0000] {local_task_job_runner.py:261} INFO - Task exited with return code 1
[2024-08-24T12:19:00.565+0000] {taskinstance.py:3891} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-08-24T12:19:00.569+0000] {local_task_job_runner.py:240} INFO - ::endgroup::
